Assets/Scripts/PlayerMouse.cs

```diff
@@ -79,14 +79,15 @@ void SetNewFaces()
 
             foreach (BlockPlugin plugin in blockBehaviour.GetPlugins())
             {
-                MoveablePlugin moveable = plugin as MoveablePlugin;
-                if (moveable == null) continue;
+                IDisplaceable displaceable = plugin as IDisplaceable;
 
-                if (!moveable.IsDisplaced() ||
-                    blockFaceOfNeighbouringBlock == moveable.GetDisplacedFace())
+                if (displaceable == null || !displaceable.CanMove()) continue;
+
+                if (!displaceable.IsDisplaced() ||
+                    blockFaceOfNeighbouringBlock == displaceable.GetDisplacedFace())
                 {
-                    BlockFace moveableFace = moveable.IsDisplaced()
-                        ? moveable.GetDisplacedFace()
+                    BlockFace moveableFace = displaceable.IsDisplaced()
+                        ? displaceable.GetDisplacedFace()
                         : blockFaceOfNeighbouringBlock;

```

Comment:
[1/2]
I realised this check is useless. Lets remove it. Line 86 - 87 already makes it such that `moveableFace` will always be `blockFaceOfNeighbouringBlock`

Reply:
[2/2]
We can then simply line `86 - 87` by creating another Method in `IDisplaceable`, something along the lines of 
```cs
// returns true if block can be displaced in the face direction provided
bool CanBeDisplaceInFaceDir(BlockFace)
```
Which does the appropriate check for moveable and disintegrate.

At the same time, we don't need `IDisplaceable#isDisplaced()`, `IDisplaceable#CanMove()`, `IDisplaceable#GetDisplacedFace()` any more with this change.